tracing/src/test/java/org/mule/test/components/tracing/ExportConfigurationChangeTestCase.java

```diff
@@ -0,0 +1,218 @@
+import static java.lang.System.setProperty;
+import static java.lang.Thread.sleep;
+import static java.nio.file.Files.copy;
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+
+import static com.linecorp.armeria.common.HttpResponse.from;
+import static com.linecorp.armeria.common.HttpStatus.OK;
+import static com.linecorp.armeria.common.HttpStatus.REQUEST_TIMEOUT;
+
+import org.mule.functional.junit4.MuleArtifactFunctionalTestCase;
+import org.mule.runtime.tracer.api.sniffer.CapturedExportedSpan;
+import org.mule.tck.probe.JUnitProbe;
+import org.mule.tck.probe.PollingProber;
+import org.mule.test.infrastructure.profiling.tracing.SpanTestHierarchy;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.server.AbstractHttpService;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.testing.junit4.server.ServerRule;
+import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+import org.jetbrains.annotations.NotNull;
+import org.junit.After;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+@Feature(PROFILING)
+@Story(DEFAULT_CORE_EVENT_TRACER)
+public class ExportConfigurationChangeTestCase extends
+    MuleArtifactFunctionalTestCase implements OpenTelemetryTracingTestRunnerConfigAnnotation {
+
+  private static final String EXPECTED_FLOW_SPAN_NAME = "mule:flow";
+  private static final String EXPECTED_SET_PAYLOAD_SPAN_NAME = "mule:set-payload";
+  private static final String FLOW_LOCATION = "flow";
+
+  private static final String TEST_ARTIFACT_ID = "ExportConfigurationChangeTestCase#test";
+
+  public static final int TIMEOUT_MILLIS = 30000;
+
+  private static final int POLL_DELAY_MILLIS = 100;
+  public static final int MAX_BACKOFF_ATTEMPTS = 2;
+
+  private static final String SET_PAYLOAD_LOCATION = "flow/processors/0";
+  private Path configFile;
+  private File file;
+
+  @Override
+  protected String getConfigFile() {
+    return "tracing/export-configuration-change.xml";
+  }
+
+  @ClassRule
+  public static final TestServerRule originalServer = new TestServerRule();
+
+  @ClassRule
+  public static final TestServerRule afterConfigurationChangeServer = new TestServerRule();
+
+  @Override
+  protected void doSetUpBeforeMuleContextCreation() throws Exception {
+    file = createTempFile("tracing", "test");
+    configFile = Paths.get(getClass().getResource("/tracing/config/exporter.conf").getPath());
```

Comment:
Will this work on Windows too?

Reply:
Changed to make sure that this works in all testing envs.